pkg/kv/kvserver/mvcc_gc_queue.go

```diff
@@ -501,14 +501,16 @@ func makeMVCCGCQueueScoreImpl(
 		r.FinalScore++
 	}
 
+	if shouldRunGC(hint, gcTTL, now) {
+		r.ShouldQueue = true
```

Comment:
I don't have a good feel whether this should instead be:
```golang
r.ShouldQueue = canAdvanceGCThreshold
```

Some code above sets `ShouldQueue` unconditionally, even if `canAdvanceGCThreshold` is false, some doesn't.

Reply:
This should be canAdvanceGCThreshold, I think. The other callers above that set it unconditionally do it to clean up intents and abort spans for abandoned transactions, which can always be removed regardless of the GC threshold.

Reply:
Done.